activejob/lib/active_job/arguments.rb

```diff
@@ -76,9 +107,10 @@ def serialize_argument(argument)
         when ActiveSupport::HashWithIndifferentAccess
           serialize_indifferent_hash(argument)
         when Hash
-          symbol_keys = argument.each_key.grep(Symbol).map(&:to_s)
+          symbol_keys = argument.each_key.grep(Symbol).map!(&:to_s)
           result = serialize_hash(argument)
           result[SYMBOL_KEYS_KEY] = symbol_keys
+          result[RUBY2_KEYWORDS_KEY] = true if Hash.ruby2_keywords_hash?(argument)
```

Comment:
This mean storing 25 characters, ~96 more bytes to the database just to get rid of a warning on Ruby. I don't think this is a good trade off.